### PR TITLE
replace some obsolete/deprecated functions/classes in django

### LINKED ISSUE #16413: Make the easy/bulk django3.2 changes in the codebase
part of #16405 - make the following easy find&replace-ish changes:

* [x] The compatibility import of django.core.exceptions.FieldDoesNotExist in django.db.models.fields is removed.

* [x] The compatibility import of django.forms.boundfield.BoundField in django.forms.forms is removed.

* [x] The smart_text() and force_text() aliases (since Django 2.0) of smart_str() and force_str() are deprecated

* [x] django.utils.translation.ugettext(), ugettext_lazy(), ugettext_noop(), ungettext(), and ungettext_lazy() are deprecated in favor of the functions that they’re aliases for: django.utils.translation.gettext(), gettext_lazy(), gettext_noop(), ngettext(), and ngettext_lazy().

* [x] Calling django.utils.crypto.get_random_string() without a length argument is deprecated.

* [x] django.conf.urls.url() alias of django.urls.re_path() is deprecated.

### PR DESCRIPTION
**Pull request by eviljeff**
part of #16405, fixes #16413 - just the straightforward find&replace for functions/classes that work equally well in django2.2

### CODE REVIEW

File: src/olympia/addons/admin.py [hunk 1]
```diff
@@ -298,7 +297,7 @@ def wrapper(*args, **kwargs):
 
         urlpatterns = super(AddonAdmin, self).get_urls()
         custom_urlpatterns = [
-            url(
+            re_path(
```
Comment (willdurand) [inline src/olympia/addons/admin.py:300]: This isn't just a replacement, right? `url` and `re_path` don't seem to be equivalent.

**Review by willdurand — COMMENTED** [review]
Is there anything that we could tell QA to verify once this patch lands? (even if it's not exhaustive?) I'd suggest you file an issue for the functions that have been changed here too.

**Comment by willdurand** [comment]
while reviewing this patch I saw some changes so I submitted what I had in mind so far. let's me know when it's ready :)

File: src/olympia/addons/admin.py (hunk 1 above)
Comment (eviljeff) [reply src/olympia/addons/admin.py:300]: https://docs.djangoproject.com/en/2.2/ref/urls/#django.conf.urls.url

**Comment by eviljeff** [comment]
> while reviewing this patch I saw some changes so I submitted what I had in mind so far. let's me know when it's ready :)

it's finished, I just forgot to actually push the codestyle fix (duplicate import) before opening the PR

**Comment by eviljeff** [comment]
> 
> 
> Is there anything that we could tell QA to verify once this patch lands? (even if it's not exhaustive?) I'd suggest you file an issue for the functions that have been changed here too.

There isn't anything for QA to test - in this PR it's just bulk renames of functions that were already aliased and/or deprecated so 3.2 will (hopefully) minimally run.  I just checked all the checkboxes in #16405 to indicate what's been addressed but I can copy those into a new issue if you think it's necessary.

**Comment by willdurand** [comment]
> There isn't anything for QA to test - in this PR it's just bulk renames of functions that were already aliased and/or deprecated so 3.2 will (hopefully) minimally run.

Not even a quick sanity check?

> I just checked all the checkboxes in #16405 to indicate what's been addressed but I can copy those into a new issue if you think it's necessary.

Yep, that would be nice. #16405 looks like a meta-issue that can stay opened as long as it is needed.

**Comment by eviljeff** [comment]
> > There isn't anything for QA to test - in this PR it's just bulk renames of functions that were already aliased and/or deprecated so 3.2 will (hopefully) minimally run.
> 
> Not even a quick sanity check?

Nothing specific I can think of, but feel free to suggest something in the issue

> > I just checked all the checkboxes in #16405 to indicate what's been addressed but I can copy those into a new issue if you think it's necessary.
> 
> Yep, that would be nice. #16405 looks like a meta-issue that can stay opened as long as it is needed.

I logged #16413

**Review by willdurand — APPROVED** [review]
Thanks!

**Pull request merged**